src/Npgsql/NpgsqlConnection.cs

```diff
@@ -926,7 +926,7 @@ async Task CloseAsync(bool async)
                 }
                 else
                 {
-                    if (_dataSource is ClusterAwareDataSource || _dataSource is TopologyAwareDataSource)
+                    if (_dataSource is ClusterAwareDataSource)// || _dataSource is TopologyAwareDataSource)
```

Comment:
This condition will be true for both `ClusterAwareDataSource` as well as `TopologyAwareDataSource` because `TopologyAwareDataSource` extends `ClusterAwareDataSource`. Removed the commented out code